src/test/shared/sam/cli/samCliValidationNotification.test.ts

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+
+import {
+    makeSamCliValidationNotification,
+    SamCliValidationNotification,
+    SamCliValidationNotificationAction
+} from '../../../../shared/sam/cli/samCliValidationNotification'
+import {
+    InvalidSamCliError,
+    InvalidSamCliVersionError,
+    SamCliNotFoundError,
+    SamCliVersionValidation,
+    SamCliVersionValidatorResult
+} from '../../../../shared/sam/cli/samCliValidator'
+
+describe('makeSamCliValidationNotification', async () => {
+    const fakeSamCliValidationNotification: SamCliValidationNotification = {
+        show: () => { throw new Error('show is unused') }
+    }
+    const actionLabelUpdateSamCli = 'Get SAM CLI'
+    const actionLabelUpdateToolkit = 'Update AWS Toolkit'
+
+    it('handles SamCliNotFoundError', async () => {
+        makeSamCliValidationNotification(
+            new SamCliNotFoundError(),
+            (message: string, actions: SamCliValidationNotificationAction[]): SamCliValidationNotification => {
+                assert.ok(
+                    message.indexOf('Unable to find SAM CLI') !== -1,
+                    `unexpected validation message: ${message}`
+                )
+                assert.strictEqual(actions.length, 1, 'unexpected action count')
+                assert.strictEqual(
+                    actions[0].label,
+                    actionLabelUpdateSamCli,
+                    `unexpected action label: ${actions[0].label}`
+                )
+
+                return fakeSamCliValidationNotification
+            }
+        )
+    })
+
+    const versionValidationTestScenarios = [
+        {
+            situation: 'SAM CLI Version is too low',
+            versionValidation: SamCliVersionValidation.VersionTooLow,
+            messageFragment: 'update your SAM CLI',
```

Comment:
Can you match on the localized string for consistency?

Reply:
The tests do not use the localization system. I chose not to export the strings (like `RECOMMENDATION_UPDATE_SAM_CLI`) from samCliValidationNotification.ts because there is no need for other non-test code to have access to it at this time.